### ISSUE TITLE
Can't read devfile registry index on github

### ISSUE THREAD

**Issue by makandre** [eclipse-che/che#14591]
### Describe the bug

I deployed Che with a pointer to an external devfile registry URL (on GitHub): https://raw.githubusercontent.com/kabanero-io/codewind-templates/master (the dashboard would append `/devfiles/index.json` onto it, which gives https://raw.githubusercontent.com/kabanero-io/codewind-templates/master/devfiles/index.json)

However, the contents could not be read:

<img width="1191" alt="image" src="https://user-images.githubusercontent.com/17596240/65193482-46fd8b80-da49-11e9-9489-6371a39d61be.png">

The console shows an error:

```
/dashboard/#/create-workspace:1 Access to XMLHttpRequest at 'https://raw.githubusercontent.com/kabanero-io/codewind-templates/master/devfiles/index.json' from origin 'http://che-che.9.21.104.221.nip.io' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status.
```

I'm fairly sure this used to work (I recall testing with 7.0.0-RC4 I believe)

### Che version

 - [x] latest
 - [ ] nightly
 - [ ] other: please specify

### Runtime
   - [ ] kubernetes (include output of `kubectl version`) 
   - [x] Openshift (include output of `oc version`)

```
oc version
oc v3.11.0+0cbc58b
kubernetes v1.11.0+d4cacc0
features: Basic-Auth

Server https://makandre-master.fyre.ibm.com:8443
openshift v3.11.0+739ab57-285
kubernetes v1.11.0+d4cacc0
```

   - [ ] minikube (include output of `minikube version` and `kubectl version`)
   - [ ] minishift (include output of `minishift version` and `oc version`)
   - [ ] docker-desktop + K8S (include output of `docker version` and `kubectl version`)
   - [ ] other: (please specify)

### Installation method
   - [x] chectl 
   - [ ] che-operator 
   - [ ] minishift-addon
   - [ ] I don't know

### Environment
   - [ ] my computer 
       - [ ] Windows
       - [ ] Linux
       - [x] macOS
   - [ ] Cloud
       - [ ] Amazon
       - [ ] Azure
       - [ ] GCE
       - [ ] other (please specify) 
   - [ ] other: please specify

**Comment by tolusha** [comment]
I don't think it should be a raw content.
From my understanding it is https://github.com/kabanero-io/codewind-templates/tree/master/devfiles
@sleshchenko Could you have a look pls

**Comment by sleshchenko** [comment]
The configured devfile registry URL (https://raw.githubusercontent.com/kabanero-io/codewind-templates/master) seems to be right. Here is another issue with cross-domain requests (CORS) it's what error message says. Looks like a bug.
cc @skabashnyuk

**Comment by skabashnyuk** [comment]
@makandre  I'm not sure that we support a feature that you've described. Why do you think it should work?

**Comment by makandre** [comment]
Hi @skabashnyuk, this was an enhancement that we had requested in the past and it was implemented in https://github.com/eclipse/che/issues/13634

**Comment by makandre** [comment]
Hi @sleshchenko or @skabashnyuk, is there an update on this issue?

**Comment by skabashnyuk** [comment]
@makandre no updates. Can you review these changes https://github.com/eclipse/che/pull/13645 they can be related to the code that supports feature that you've mentioned and suggest a patch?

**Comment by makandre** [comment]
@skabashnyuk actually I had worked with @tobespc on the changes in #13645 to enable reading of the devfile registry index from GitHub.  At the time it was delivered I verified that it worked as intended.  So I'm not sure what changed since then but now it's throwing the CORS error which I reported above.

**Comment by skabashnyuk** [comment]
Strange. Github response contains ```Access-Control-Allow-Origin```
```
curl -v https://raw.githubusercontent.com/kabanero-io/codewind-templates/master/devfiles/index.json

> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Security-Policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
< Strict-Transport-Security: max-age=31536000
< X-Content-Type-Options: nosniff
< X-Frame-Options: deny
< X-XSS-Protection: 1; mode=block
< ETag: "8e01a130e4a56d109cb78c8ccac1ecf38e79662a"
< Content-Type: text/plain; charset=utf-8
< Cache-Control: max-age=300
< X-Geo-Block-List:
< X-GitHub-Request-Id: 648E:1963:BF0E3:F7BB5:5D93558A
< Content-Length: 2300
< Accept-Ranges: bytes
< Date: Tue, 01 Oct 2019 13:32:59 GMT
< Via: 1.1 varnish
< Connection: keep-alive
< X-Served-By: cache-ams21047-AMS
< X-Cache: MISS
< X-Cache-Hits: 0
< X-Timer: S1569936779.441782,VS0,VE174
< Vary: Authorization,Accept-Encoding
< Access-Control-Allow-Origin: *
< X-Fastly-Request-ID: 7ef578b63a42a723a732b222ea7ec4974b9cba70
< Expires: Tue, 01 Oct 2019 13:37:59 GMT
< Source-Age: 0
<
```
Can you show your browser developer console .  We should see preflight request https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request. Maybe it's some howe blocked in your environment?

**Comment by makandre** [comment]
You may be right could be an environmental issue... I'll investigate further

**Comment by sleshchenko** [comment]
@makandre Seems this issue is not actual anymore, see https://youtu.be/Ce3pXZ4cOTw Feel free to reopen if it still does not work for you.
@akurinnoy Could you confirm that you used GitHub repo as a plugin registry?

**Comment by akurinnoy** [comment]
@sleshchenko yes, that's correct
